pkg/catalogsourceconfig/handler.go

```diff
@@ -0,0 +1,143 @@
+package catalogsourceconfig
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	opsrc "github.com/operator-framework/operator-marketplace/pkg/operatorsource"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Copied from https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/registry/configmap_loader.go#L18
+// TBD: Vendor in the folder once we require more than just these constants from the OLM registry code
+const (
+	ConfigMapCRDName     = "customResourceDefinitions"
+	ConfigMapCSVName     = "clusterServiceVersions"
+	ConfigMapPackageName = "packages"
+)
+
+// Handler is the interface that wraps the Handle method
+type Handler interface {
+	Handle(ctx context.Context, event sdk.Event) error
```

Comment:
I think we can have `Handle` specialized it here, passing csc instead of ctx as it's not adding much of value

Reply:
I think we will need the parent `ctx` if we want to handle cancellation or graceful shutdown. That's probably the reason sdk is passing it along to the base `stub.Handler` function.

Reply:
Passing it to the base function is fine, I think, we can always catch the error in the calling function and we handle graceful shutdown there. The main problem I see with this approach is explicit type conversion which we already converted in calling function. This is something beyond the scope of the current PR.